wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -253,21 +265,36 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
   /** If the given UTXO is marked as unspent, updates
     * its spending status. Otherwise returns `None`.
     */
-  private def markAsPendingSpent(
-      out: SpendingInfoDb): Future[Option[SpendingInfoDb]] = {
+  private def markAsSpent(
+      out: SpendingInfoDb,
+      spendingTxId: DoubleSha256DigestBE): Future[Option[SpendingInfoDb]] = {
     out.state match {
       case TxoState.ConfirmedReceived | TxoState.PendingConfirmationsReceived |
           TxoState.ImmatureCoinbase =>
         val updated =
-          out.copyWithState(state = TxoState.PendingConfirmationsSpent)
+          out
+            .copyWithState(state = TxoState.PendingConfirmationsSpent)
+            .copyWithSpendingTxId(spendingTxId)
         val updatedF =
           spendingInfoDAO.update(updated)
         updatedF.foreach(updated =>
           logger.debug(
             s"Marked utxo=${updated.toHumanReadableString} as state=${updated.state}"))
         updatedF.map(Some(_))
-      case TxoState.Reserved | TxoState.ConfirmedSpent |
-          TxoState.PendingConfirmationsSpent | TxoState.DoesNotExist =>
+      case TxoState.Reserved | TxoState.PendingConfirmationsSpent =>
```

Comment:
Again, i think this is a problem. If I understand correctly, we saying our wallet behavior is to "only keep track of the _last seen_ spending transaction for an output. Disregard previous spending transactions associated with this utxo" 

At minimum, we should be logging `warn` messages for this. 

If I understand correctly, we do need to "pick one" (first seen spending tx, or last seen spending tx). If this is the decision we want to make until #2179 is done, let's make a unit test for it. 

unfortunately i don't think there is any guarantees we can provide that the last seen transaction wasn't like 10 blocks behind the tip is there?

Do we have unit tests for an old spending transaction being overwritten by a new spending transaction? If so could you link me?